honeytrap: initial run error

When starting the initial run, honeytrap crashes. This is due to a fault in creation of the `.honeytrap` folder and its contents in the user home. Second run after crash is successful.

```
$ ./honeytrap
2018/04/24 09:36:02 Using config file config.toml
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xb4f866]

goroutine 1 [running]:
github.com/honeytrap/honeytrap/server.New(0xc0421a8ba0, 0x3, 0x4, 0x0, 0x3, 0xc0421a8ba0)
        C:/Projects/Go/src/github.com/honeytrap/honeytrap/server/honeytrap.go:133 +0x156
github.com/honeytrap/honeytrap/cmd/honeytrap.serve(0xc0421e0000, 0xc0421e0000, 0xc042075d37)
        C:/Projects/Go/src/github.com/honeytrap/honeytrap/cmd/honeytrap/main.go:170 +0x507
github.com/honeytrap/honeytrap/vendor/gopkg.in/urfave/cli%2ev1.HandleAction(0xbf8e80, 0xd62e10, 0xc0421e0000, 0xc042194420, 0x0)
        C:/Projects/Go/src/github.com/honeytrap/honeytrap/vendor/gopkg.in/urfave/cli.v1/app.go:490 +0xd9
github.com/honeytrap/honeytrap/vendor/gopkg.in/urfave/cli%2ev1.(*App).Run(0xc0421dc000, 0xc04204a160, 0x1, 0x1, 0x0, 0x0)
        C:/Projects/Go/src/github.com/honeytrap/honeytrap/vendor/gopkg.in/urfave/cli.v1/app.go:264 +0x63c
main.main()
        C:/Projects/Go/src/github.com/honeytrap/honeytrap/main.go:47 +0x52
```

--- a/server/options.go
+++ b/server/options.go
@@ -110,8 +110,9 @@ func WithDataDir(s string) (OptionFn, error) {
 			if err != nil {
 				return nil, err
 			}
+		} else {
+			return nil, err
 		}
-		return nil, err
 	}
 
 	return func(b *Honeytrap) error {

--- a/server/options.go
+++ b/server/options.go
@@ -103,16 +103,15 @@ func WithDataDir(s string) (OptionFn, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	_, err = os.Stat(p)
-	if err != nil {
-		if os.IsNotExist(err) {
-			err = os.Mkdir(p, 0755)
-			if err != nil {
-				return nil, err
-			}
-		} else {
+	if os.IsNotExist(err) {
+		err = os.Mkdir(p, 0755)
+		if err != nil {
 			return nil, err
 		}
+	} else if err != nil {
+		return nil, err
 	}
 
 	return func(b *Honeytrap) error {

